app/main.py

```diff
@@ -85,6 +93,30 @@ def create_app() -> FastAPI:
     return app
 
 
+def get_auth_deps() -> list[Depends]:
+    "function in order to fetch dependencies for authentication"
+    deps = os.environ.get("OTEAPI_AUTH_DEPS")
+    if deps:
+        mods = [
+            module.replace(" ", str()).rpartition(".") for module in deps.split("|")
+        ]
+        try:
+            imports = [
+                getattr(import_module(module), classname)
+                for (module, dot, classname) in mods
```

Comment:
I think, traditionally, one writes out the module as you suggest (`package_name.module.sub_module`), but class names are added after a colon, e.g., `package_name.module.sub_module:MyClass`.
So I'd probably go this route instead?

Also, using `rpartition` is nice, but dangerous. What if the class is imported from the root module/package name? I.e., `package_name:MyClass`? Then your import statement would be `getattr(import_module(""), "package_name")` in the current version. I think with the colon this should all work out, as you can `rpartition` with the colon and all this should be fine :)

Also:
```suggestion
                getattr(import_module(module), classname)
                for (module, _, classname) in modules
```

Reply:
Very good hint. I will apply your suggestion with the colon